ports/nrf/common-hal/microcontroller/Processor.c

```diff
@@ -136,6 +138,12 @@ mcu_reset_reason_t common_hal_mcu_processor_get_reset_reason(void) {
         r = RESET_REASON_WATCHDOG;
     } else if (reset_reason_saved & POWER_RESETREAS_SREQ_Msk) {
         r = RESET_REASON_SOFTWARE;
+        // Our "deep sleep" is still actually light sleep followed by a software
+        // reset. Adding this check here ensures we treat it as-if we're waking
+        // from deep sleep.
+        if (sleepmem_wakeup_event != SLEEPMEM_WAKEUP_BY_NONE) {
```

Comment:
Build failure with `undefined reference to 'sleepmem_wakeup_event'` on some boards.

Reply:
The BlueMicro840 built fine. Tested it with 5 and 4 encoders.
5th initialization fails and `RuntimeError: All channels in use` exception comes out as expected.

With 4 encoders only, all 4 work as expected.

Haven't tested with using other GPIOTE consumers but I expect rotaryio will be fine if the above test is good.